### PR TITLE
:sparkles: Feature(comment): add vssue comment system

### PR DESCRIPTION
**Pull request by LocTran016**
Add Vssue comment system (support to store comments in gitlab, github and some other platforms)

ISSUES CLOSED: #297

### CODE REVIEW

File: layout/includes/comments/vssue.pug [hunk 1]
```diff
@@ -0,0 +1,26 @@
+if theme.vssue && theme.vssue.enable
+  #vssue-container
+  link(rel="stylesheet" href="https://unpkg.com/vssue/dist/vssue.min.css")
```
Comment (Molunerfinn) [inline layout/includes/comments/vssue.pug:3]: use jsdelivr's CDN

**Review by Molunerfinn — CHANGES_REQUESTED** [review]

File: layout/includes/comments/vssue.pug (hunk 1 above)
Comment (LocTran016) [reply layout/includes/comments/vssue.pug]: Changed

File: layout/includes/comments/vssue.pug [hunk 2]
```diff
@@ -1,9 +1,9 @@
 if theme.vssue && theme.vssue.enable
   #vssue-container
-  link(rel="stylesheet" href="https://unpkg.com/vssue/dist/vssue.min.css")
-  script(src="https://unpkg.com/vue/dist/vue.min.js")
+  link(rel="stylesheet" href="https://cdn.jsdelivr.net/npm/vssue@1.4.6/dist/vssue.css")
+  script(src="https://cdn.jsdelivr.net/npm/vue@2.6.12/dist/vue.min.js")
   - var platform = theme.vssue.platform || 'github'
-  script(src="https://unpkg.com/vssue/dist/vssue." + platform + ".min.js")
+  script(src="cdn.jsdelivr.net/npm/vssue@1.4.6/dist/vssue." + platform + ".min.js")
```
Comment (Molunerfinn) [inline layout/includes/comments/vssue.pug:6]: https?

**Review by Molunerfinn — CHANGES_REQUESTED** [review]

File: layout/includes/comments/vssue.pug (hunk 2 above)
Comment (LocTran016) [reply layout/includes/comments/vssue.pug]: Fixed below

File: layout/includes/comments/vssue.pug [hunk 3]
```diff
@@ -0,0 +1,35 @@
+doctype html
```
Comment (Molunerfinn) [inline layout/includes/comments/vssue.pug:1]: ??? unnecessary, don't add too many unnecessary codes.

**Review by Molunerfinn — CHANGES_REQUESTED** [review]

**Comment by Molunerfinn** [comment]
If you have finished, just comment & I will review & merge in time. I have work to do, so I can't merge in time, but just be patient & don't add unnecessary codes, thanks

File: layout/includes/comments/vssue.pug (hunk 3 above)
Comment (Molunerfinn) [reply layout/includes/comments/vssue.pug:1]: It's a component, not a page, don't add these, just follow other comment components.

**Comment by LocTran016** [comment]
Sorry, in the next few days I have a important exam at my school so this will be delayed for some days. Sorry for this

**Review by Molunerfinn — CHANGES_REQUESTED** [review]

**Comment by Molunerfinn** [comment]
> Sorry, in the next few days I have a important exam at my school so this will be delayed for some days. Sorry for this

Come on, I wish you a good result in the exam!

**Comment by LocTran016** [comment]
> > Sorry, in the next few days I have a important exam at my school so this will be delayed for some days. Sorry for this
> 
> Come on, I wish you a good result in the exam!

Thanks.
I've changed the code now, maybe you could review this

File: layout/includes/comments/vssue.pug [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  - var platform = theme.vssue.platform || 'github'
+  script(src="https://cdn.jsdelivr.net/npm/vssue@1.4.6/dist/vssue." + platform + ".min.js")
+  //- Used for scripts
+  - var perPage = theme.vssue.perPage || 10
+  - var autoCreateIssue = theme.vssue.autoCreateIssue || false
+  if theme.vssue.baseURL != ""
+    - var baseURL = theme.vssue.baseURL
+  else if ['github-v4', 'github'].includes(theme.vssue.platform)
+    - var baseURL = 'https://github.com'
+  else if theme.vssue.platform = 'gitlab'
```
Comment (Molunerfinn) [inline layout/includes/comments/vssue.pug:14]: use `===` not `=`, `=` is assignment, not comparison.

**Review by Molunerfinn — CHANGES_REQUESTED** [review]

File: layout/includes/comments/vssue.pug (hunk 4 above)
Comment (Molunerfinn) [reply layout/includes/comments/vssue.pug:14]: not `theme.vssue.platform = 'gitlab'` but `theme.vssue.platform === 'gitlab'`

Comment (LocTran016) [reply layout/includes/comments/vssue.pug]: I forgot to fix this. Sorry

**Comment by Molunerfinn** [comment]
Thanks for your contribution~

**Pull request merged**